src/tcpinfo_lib.cc

```diff
@@ -0,0 +1,63 @@
+// Copyright 2016 measurement-lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "tcpinfo_lib.h"
+
+namespace mlab {
+namespace netlink {
+
+TCPDiagnosticsProto TCPInfoParser::ParseNLMsg(
+    const struct nlmsghdr* msg, int protocol) const {
+  TCPDiagnosticsProto proto;
+  NLMsgToProto(msg, protocol, &proto);
+  return std::move(proto);
+}
+
+void TCPInfoParser::NLMsgToProto(const struct nlmsghdr* nlh, int protocol,
+                                 TCPDiagnosticsProto* proto) const {
+}
+
+bool ConnectionFilter::Accept(
+    const mlab::netlink::InetSocketIDProto& socket) const {
+  return true;
+}
+bool ConnectionFilter::Accept(const struct nlmsghdr* msg) const {
+  return true;
+}
+
+void TCPInfoPoller::PollOnce() {}
```

Comment:
Are these empty function skeletons waiting to be filled in later, or are they default implementations, left blank so that subclasses can choose what subset of the functionality they might want to implement? If the second, maybe define all the empty functions inline in the .h file? If the first, all is good here.